Azure/sonic-utilities
# [tests] Do not trim from PATH if we did not append to it; Clean up/fix shebangs in scripts

**- What I did**

Some tests were trimming the last item from system PATH, but they never appended anything to it, thus these tests were trimming valid entries from the tail of the system path, which in turn caused scripts with shebangs like `#!/usr/bin/env python` to fail, because once `/usr/bin` was removed from PATH, `python` could not be found.

**- How I did it**

- Remove lines which trimmed the last entry from system PATH in the teardown function if nothing was appended to PATH in the setup method.
- Remove shebangs from non-executable Python files
- Fix shebangs which had a space in them
- Replace `#!/usr/bin/python` shebangs with the preferred, more portable `#!/usr/bin/env python` in executable scripts

**- How to verify it**

Ensure scripts with `#!/usr/bin/env python` shebangs execute properly in pytests.

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
